lib/internal/Magento/Framework/View/Element/AbstractBlock.php

```diff
@@ -650,10 +657,17 @@ protected function _beforeToHtml()
     public function toHtml()
     {
         $this->_eventManager->dispatch('view_block_abstract_to_html_before', ['block' => $this]);
-        if ($this->_scopeConfig->getValue(
-            'advanced/modules_disable_output/' . $this->getModuleName(),
-            \Magento\Store\Model\ScopeInterface::SCOPE_STORE
-        )) {
+
+        if (static::$disabledModules === null || !isset(static::$disabledModules[$this->getModuleName()])) {
+            static::$disabledModules[$this->getModuleName()] = $this->_scopeConfig->getValue(
+                'advanced/modules_disable_output/' . $this->getModuleName(),
+                \Magento\Store\Model\ScopeInterface::SCOPE_STORE
+            ) ?? [];
+        }
```

Comment:
Why can't we retrieve all the disabled modules at once, as it was suggested originally?

Reply:
That was the original thought out method indeed.